### PR TITLE
Allow fields to be opt-in rather than opt-out

### PR DESCRIPTION
**Pull request by theuni**
This addresses the integration concern seen here: https://github.com/theuni/bitcoin/blob/minisketch-split/src/Makefile.minisketch.include#L4

If a field is explicitly enabled and disabled, disable takes precedence.
Defines are intended to work as such:
```
Default: All fields enabled
-DDISABLE_FIELD_3: All fields except 3 are enabled
-DENABLE_FIELD_3: All fields enabled
-DDISABLE_DEFAULT_FIELDS: Logical error, no fields enabled
-DDISABLE_DEFAULT_FIELDS -DENABLE_FIELD_3: Only field 3 enabled
-DDISABLE_DEFAULT_FIELDS -DENABLE_FIELD_2 -DENABLE_FIELD_3: Fields 2 and 3 are enabled
-DDISABLE_DEFAULT_FIELDS -DENABLE_FIELD_2 -DENABLE_FIELD_3 -DDISABLE_FIELD_3: Only field 2 enabled
```

### CODE REVIEW

**Comment by sipa** [comment]
I think it would be cleaner to have a layer (in one of the .h) files which takes the "external" defines (which can be both opt-in or opt-out based), and define separate internal defines which are opt-in only. That would avoid littering the "explicitly enabled or default enabled and not disabled" logic everywhere.

**Comment by theuni** [comment]
Aha, agreed. Will give that a go.

**Comment by theuni** [comment]
Done. The resulting header is big and ugly, but it's self-contained and fairly straightforward I think.

**Comment by theuni** [comment]
CI failures appear to be unrelated.

**Comment by sipa** [comment]
utACK faff8a01353cd36c1abc89c79d38287624f59a9c

**Comment by fanquake** [comment]
> CI failures appear to be unrelated.

The issue is that brew is trying to use bintray, which no longer exists. Could be solved by either running a `brew update` step before installing packages (may take some time to run), or using a newer macOS image which would have a more up-to-date version of brew already available.

**Comment by sipa** [comment]
@fanquake Feel like PR'ing a fix?

**Comment by fanquake** [comment]
> @fanquake Feel like PR'ing a fix?

Done in #48.

File: doc/gen_params.sage [hunk 1]
```diff
@@ -315,14 +315,14 @@ def print_result(bits, style):
...
-    print("#ifndef DISABLE_FIELD_%i" % bits)
+    print("#ifdef ENABLE_FIELD_INT_%i" % bits)
     print("// %i bit field" % bits)
     print_result(bits, INT)
     print("#endif")
     print("")
 
 for bits in range(2, 65):
-    print("#ifndef DISABLE_FIELD_%i" % bits)
+    print("#ifdef ENAABLE_FIELD_INT_%i" % bits)
```
Comment (jnewbery) [inline doc/gen_params.sage:325]:
```suggestion
    print("#ifdef ENABLE_FIELD_INT_%i" % bits)
```

**Comment by sipa** [comment]
Squash the squashme?

**Comment by sipa** [comment]
utACK 67f5160158e7bbe2810ca1c80b250e0254a7c63f

**Pull request merged**